#ifndef MAINWINDOW_H
#define MAINWINDOW_H

#include <QMainWindow>
#include <QTimer>
#include "NGLScene.h"
#include "JelloCube.h"

namespace Ui {
	class MainWindow;
}

class MainWindow : public QMainWindow
{
	Q_OBJECT

	public:
		explicit MainWindow(QWidget *parent = 0);
		~MainWindow();

	public slots:
		//----------------------------------------------------------------------------------------------------------------------
		/// @brief reset our jello cube and update some gui things
		//----------------------------------------------------------------------------------------------------------------------
		void resetJelloCube();
		//----------------------------------------------------------------------------------------------------------------------
		/// @brief toggle the sphere collider
		//----------------------------------------------------------------------------------------------------------------------
		void toggleSphereCollider();
		//----------------------------------------------------------------------------------------------------------------------
		/// @brief update FPS label
		//----------------------------------------------------------------------------------------------------------------------
		void updateFPS();

	private:
		// QWidget *m_ui;
		Ui::MainWindow *m_ui;
		//----------------------------------------------------------------------------------------------------------------------
		/// @brief our openGL widget
		//----------------------------------------------------------------------------------------------------------------------
		NGLScene *m_gl;
		//----------------------------------------------------------------------------------------------------------------------
		/// @brief our jello cube
		//----------------------------------------------------------------------------------------------------------------------
		JelloCube *m_jelloCube;
		//----------------------------------------------------------------------------------------------------------------------
		/// @brief timer object for polling
		//----------------------------------------------------------------------------------------------------------------------
		QTimer *m_timer;
		//----------------------------------------------------------------------------------------------------------------------
		/// @brief update the ui text
		//----------------------------------------------------------------------------------------------------------------------
		void updateUIText();
};

#endif // MAINWINDOW_H
